### ISSUE TITLE
Refined storage system saying i dont have enough of a certain item in the system to autocraft even though i do

### ISSUE THREAD

**Issue by Senthin1** [refinedmods/refinedstorage#2866]
#### Issue description:
When i want to autocraft something that I have the materials for, the crafting preview a lot of the times says I don't have enough of the item to do the autocrafting.

#### What happens:

Trying to craft a reinforced alloy from Mekanism, 

![image](https://user-images.githubusercontent.com/78551691/106915093-a78b7800-66d3-11eb-958f-79b8a2b29db6.png)

Saying I don't have enough iron, even tho I have this much iron in the system, 

![image](https://user-images.githubusercontent.com/78551691/106915235-cab62780-66d3-11eb-926c-0c4d0af33636.png)

Here is the pattern:
![image](https://user-images.githubusercontent.com/78551691/106915500-11a41d00-66d4-11eb-8f0c-1fd7f6be339e.png)

#### What you expected to happen:
For it to craft 10 Reinforced alloy

#### Steps to reproduce:
1. make the pattern
2. put the crafter facing into a chest which then feeds into an infusing factory
...

#### Version (make sure you are on the latest version before reporting): 

- Minecraft: 1.16.5
- Forge: 35.1.36
- Refined Storage: 1.9.11

Does this issue occur on a server? [yes/no]: yes

**Update:** So now it does not show that I need iron for the autografting and it's allowing me to craft. However, this has happened many times before and it's frustrating that it keeps coming.:

![image](https://user-images.githubusercontent.com/78551691/106920634-4c5c8400-66d9-11eb-8a3d-8fe8006e7bd3.png)

Update part 2: The problem is back :|

**Comment by Darkere** [comment]
Can you post the pattern for the infused alloy too?

**Comment by Senthin1** [comment]
> Can you post the pattern for the infused alloy too?

![image](https://user-images.githubusercontent.com/78551691/106920097-bde80280-66d8-11eb-8e44-be2e4ece1956.png)

**Comment by Darkere** [comment]
Try loading up the pattern in the pattern grid. 

Then ctrl + click the iron. Post the screen that pops up.

**Comment by Darkere** [comment]
Actually that only works if they are tagged. Check the Recipe in JEI if there are multiple types of iron ingots in your game.

**Comment by Senthin1** [comment]
> Actually that only works if they are tagged. Check the Recipe in JEI if there are multiple types of iron ingots in your game.

![image](https://user-images.githubusercontent.com/78551691/106937026-45d70800-66eb-11eb-976e-a02282fffc9d.png)

nope only 1 iron ingot

**Comment by Moorest42** [comment]
Do you have your items (the ones that aren't recognized) stored in an external inventory like a chest or drawer? I stored my items in an advanced black hole unit and my autocrafting won't recognize them even though my grid does. It worked before I stored my items externally in black hole units.

**Comment by Moorest42** [comment]
Do you have your items (the ones that aren't recognized) stored in an external inventory like a chest or drawer? I stored my items in an advanced black hole unit and my autocrafting sometimes won't recognize them even though my grid does. It worked before I stored my items externally in black hole units.

**Comment by Moorest42** [comment]
I found out that for what ever reaseon my items in the advanced black hole units get a extra tag. So I turned off the exact mode. Now it's seems to work. Maybe you have a similar problem.
![with_tag](https://user-images.githubusercontent.com/71607122/107994516-17a3d300-6fdd-11eb-99d2-0f4712311633.PNG)
![without_tag](https://user-images.githubusercontent.com/71607122/107994533-1c688700-6fdd-11eb-9577-603b0b5e89d7.PNG)

**Comment by Darkere** [comment]
@Moorest42 what pack /version are you on? We've been hunting a bug that does that for ages.

**Comment by Senthin1** [comment]
> I found out that for what ever reaseon my items in the advanced black hole units get a extra tag. So I turned off the exact mode. Now it's seems to work. Maybe you have a similar problem.
> ![with_tag](https://user-images.githubusercontent.com/71607122/107994516-17a3d300-6fdd-11eb-99d2-0f4712311633.PNG)
> ![without_tag](https://user-images.githubusercontent.com/71607122/107994533-1c688700-6fdd-11eb-9577-603b0b5e89d7.PNG)

I had this same problem and I found out that if you make the autocrafting pattern from JEI, it will put the mekanism version of the steel ingot instead of the thermal dynamics steel ingot

**Comment by Moorest42** [comment]
> 
> 
> @Moorest42 what pack /version are you on? We've been hunting a bug that does that for ages.

I am playing on a Valhelsia server with a few frinds. 
Valhelsia  3 - 1.16.5 - version 3.1.8a

**Comment by raoulvdberge** [comment]
Closing this issue as I don't really know how to move this forward. If you have any suggestions, feel free.

**Comment by Sanzath** [comment]
I just experience this issue today myself. I found that my diamonds could be stored in my refined storage system but weren't really being recognized for crafts. It turns out, some of my diamonds had empty NBT rather than no NBT at all, and the empty NBT was causing them to not work with several different mods. I still haven't figured out how they ended up with NBT in the first place.

To resolve my issue, I crafted my diamonds with NBT into blocks and back into diamonds, causing them to lose their NBT and work properly again.
